Letca_Anna/week_02/Demo/Program.cs

```diff
@@ -33,24 +43,103 @@ static async Task Main(string[] args)
                 from menuItemResult1 in RestaurantDomain.CreateAndAddMenuItem("carbonara", 25, menu)
                 from menuItemResult2 in RestaurantDomain.CreateAndAddMenuItem("conpesto", 20, menu)
                 let menuItem2=(menuItemResult2 as MenuItemAdded)?.MenuItem
-                from clientResult in RestaurantDomain.CreateClient("gucdg34u6trgfh")
-                let client = (clientResult as ClientCreated)?.Client
-                from addToCartResult in RestaurantDomain.AddItemToCart(menuItem2, client)
-                from orderResult in RestaurantDomain.CreateAndPlaceOrder(client, restaurant)
+                from clientResult in RestaurantDomain.CreateClient("gucdg34u6trgfh","Anca")
+                //from orderResult in RestaurantDomain.CreateAndPlaceOrder(client, restaurant)
                 select restaurantResult;
 
-            //remove item from cart(item, client.cart)
-            //create and place order(client, restaurant)
-
-            Console.WriteLine(expr);
-
             var interpreter = new LiveInterpreterAsync(serviceProvider);
 
             var result = await interpreter.Interpret(expr, Unit.Default);
 
             var finalResult = result.Match<bool>(OnRestaurantCreated, OnRestaurantNotCreated);
             Assert.True(finalResult);
 
+            Restaurant foundRestaurant=null;
+            Menu foundMenu = null;
+            Client foundClient = null;
+
+            do
+            {
+                PrintInputOptions();
+                input = Console.ReadLine(); 
+                switch (input)
+                {
+                    case "1": // login (suppose client is already created/registered)
+                        var getClientExpr =
+                                        from client in RestaurantDomain.GetClient("gucdg34u6trgfh")
+                                        select client;
+                        var interpretedClient = await interpreter.Interpret(getClientExpr, Unit.Default);
+                        foundClient = (interpretedClient as ClientFound).Client;
```

Comment:
This will kaboom. The main point of having that generator is to enforce the consumer (i.e. the developer that consumes that API) to cover all possible scenarios that he himself or someone else foresee. You decided that the GetClient can return ClientFound and ClientNotFound. What happens if you change your console, take the id/name of the clinet as an input and try to retrieve it? What happens if it doesn't exist? 

Can you try a fix for this using the intended Match extensions?